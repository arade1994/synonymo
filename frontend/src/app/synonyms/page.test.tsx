import { describe, expect, test, vi } from "vitest";

import mockClient, { mockPost } from "@/mocks/buildClient.mock";
import routerMock, { mockPush } from "@/mocks/router.mock";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import SynonymsPage from "./page";

vi.mock("next/navigation", () => ({
  useRouter: () => routerMock,
}));

vi.mock("@/api/buildClient", () => {
  return {
    default: () => mockClient,
  };
});

describe("SynonymsPage", () => {
  test("It should render and match a snapshot by default", () => {
    const { baseElement } = render(<SynonymsPage />);

    expect(baseElement).toMatchSnapshot();
  });

  test("It should render the home link", () => {
    render(<SynonymsPage />);

    const homeLink = screen.getByTestId("homeLink");

    expect(homeLink).toBeDefined();
  });

  test("It should render the word input", async () => {
    render(<SynonymsPage />);

    const wordInput = screen.getByTestId("wordInput");
    expect(wordInput).toBeDefined();

    await userEvent.click(wordInput);
    await userEvent.type(wordInput, "happy");
  });

  test("It should render the synonym input", async () => {
    render(<SynonymsPage />);

    const synonymInput = screen.getByTestId("synonymInput");
    expect(synonymInput).toBeDefined();

    await userEvent.click(synonymInput);
    await userEvent.type(synonymInput, "joyful");
  });

  test("It should call the post method with the correct arguments", async () => {
    render(<SynonymsPage />);

    const wordInput = screen.getByTestId("wordInput");
    const synonymInput = screen.getByTestId("synonymInput");
    const addSynonymButton = screen.getByTestId("addSynonymButton");

    await userEvent.click(wordInput);
    await userEvent.type(wordInput, "happy");
    await userEvent.click(synonymInput);
    await userEvent.type(synonymInput, "joyful");
    await userEvent.click(addSynonymButton);

    expect(mockPost).toHaveBeenCalledWith("/api/synonyms", {
      word: "happy",
      synonym: "joyful",
    });
    expect(mockPush).toHaveBeenCalledWith("/");
  });
});
