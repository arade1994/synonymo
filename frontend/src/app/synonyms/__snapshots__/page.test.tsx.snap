// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SynonymsPage > It should render and match a snapshot by default 1`] = `
<body>
  <div>
    <main
      class="_main_4ac91f"
    >
      <a
        class="_homeLink_4ac91f"
        href="/"
        id="homeLink"
      >
        ‚Üê Back to Home
      </a>
      <h1
        class="_title_4ac91f"
      >
        Add a 
        <span
          class="_highlight_4ac91f"
        >
          new
        </span>
         Synonym
      </h1>
      <form
        class="_form_4ac91f"
      >
        <div
          class="_inputGroup_4ac91f"
        >
          <input
            class="_input_4ac91f"
            id="wordInput"
            placeholder="Enter a word"
            type="text"
            value=""
          />
        </div>
        <div
          class="_inputGroup_4ac91f"
        >
          <input
            class="_input_4ac91f"
            id="synonymInput"
            placeholder="Enter a synonym"
            type="text"
            value=""
          />
        </div>
        <button
          class="_button_4ac91f"
          id="addSynonymButton"
          type="submit"
        >
          Add Synonym
        </button>
      </form>
    </main>
  </div>
</body>
`;
